build.gradle

```diff
@@ -257,3 +258,73 @@ task dependencyReport {
     }
 }
 
+// License reporting
+downloadLicenses {
+    includeProjectDependencies = true
+
+    ext.apacheTwo = license('Apache License, Version 2.0', 'http://opensource.org/licenses/Apache-2.0')
+    ext.mit = license('MIT License', 'http://www.opensource.org/licenses/mit-license.php')
+    ext.bsd = license('BSD License', 'http://www.opensource.org/licenses/bsd-license.php')
+    ext.bsd3Clause = license('BSD 3-Clause', 'http://opensource.org/licenses/BSD-3-Clause')
+
+    aliases = [
+            (apacheTwo) : [
+                    'The Apache Software License, Version 2.0', 'Apache License Version 2.0', 'Apache License, Version 2.0',
+                    'Apache 2', 'Apache 2.0', 'Apache License 2.0', 'Apache-2.0',
+                    license('Apache License', 'http://www.apache.org/licenses/LICENSE-2.0'),
+                    license('Apache Software Licenses', 'http://www.apache.org/licenses/LICENSE-2.0.txt'),
+                    license('Apache', 'http://www.opensource.org/licenses/Apache-2.0')
+            ],
+            (mit) : ['The MIT License'],
+            (bsd) : ['BSD', 'Berkeley Software Distribution (BSD) License',
+                     license('New BSD License', 'http://www.opensource.org/licenses/bsd-license.php')
+            ],
+            (bsd3Clause) : [
+                    license('BSD 3-clause', 'http://opensource.org/licenses/BSD-3-Clause'),
+                    license('BSD 3-Clause', 'http://www.scala-lang.org/license.html')
+            ]
+    ]
+
+    licenses = [
+            (group('org.apache.httpcomponents')): apacheTwo
+    ]
+}
+
+task verifyLicenses {
+    description "Verify that all dependencies use white-listed licenses."
+    dependsOn ':downloadLicenses'
+
+    ext.whiteList = [
```

Comment:
Addressed with `dependencyConfiguration = 'compile'` in `downloadLicenses`. Unfortunately we still have logback with LGPL and `mysql-connector-java` with GPL (latter one used for anorm).